yb-voyager/src/query/queryissue/detectors.go

```diff
@@ -391,6 +394,54 @@ func (d *JsonQueryFunctionDetector) GetIssues() []QueryIssue {
 	return issues
 }
 
+type UniqueNullsNotDistinctDetector struct {
+	query    string
+	detected bool
+}
+
+func NewUniqueNullsNotDistinctDetector(query string) *UniqueNullsNotDistinctDetector {
+	return &UniqueNullsNotDistinctDetector{
+		query: query,
+	}
+}
+
+// Detect checks if a unique constraint is defined which has nulls not distinct
+func (d *UniqueNullsNotDistinctDetector) Detect(msg protoreflect.Message) error {
+	if queryparser.GetMsgFullName(msg) == queryparser.PG_QUERY_INDEX_STMT_NODE {
+		indexStmt, err := queryparser.ProtoAsIndexStmt(msg)
+		if err != nil {
+			return err
+		}
+
+		if indexStmt.Unique && indexStmt.NullsNotDistinct {
+			d.detected = true
+		}
+	} else if queryparser.GetMsgFullName(msg) == queryparser.PG_QUERY_TABLECONSTRAINT_NODE {
+		proto, ok := msg.Interface().(proto.Message)
+		if !ok {
+			return fmt.Errorf("failed to cast msg to proto.Message")
+		}
+		constraintNode, ok := proto.(*pg_query.Constraint)
+		if !ok {
+			return fmt.Errorf("failed to cast msg to %s", queryparser.PG_QUERY_TABLECONSTRAINT_NODE)
+		}
+
+		if constraintNode.Contype == pg_query.ConstrType_CONSTR_UNIQUE && constraintNode.NullsNotDistinct {
```

Comment:
define pg_query.ConstrType_CONSTR_UNIQUE in queryparser pkg. 
Again, high level goal is to not import pg_query pkg in queryissue as much as possible; keep all the code related to parser inside queryparser pkg.

Reply:
Done